pkg/controller/storagecluster/reconcile.go

```diff
@@ -931,3 +915,24 @@ func remove(slice []string, s string) (result []string) {
 	}
 	return
 }
+
+func getMonCount(nodeCount int) int {
+	count := defaults.MonCountMin
+
+	// return static value if overriden
+	override := os.Getenv(monOverrideEnvVar)
+	if override != "" {
+		count, err := strconv.Atoi(override)
+		if err != nil {
+			log.Error(err, "could not decode env var %s", monOverrideEnvVar)
+		} else {
+			return count
+		}
+	}
+
+	if nodeCount >= defaults.MonCountMax {
+		count = defaults.MonCountMax
```

Comment:
If I read this correctly, it will result in four mons for a 4-node Ceph cluster, which will be more error-prone than 3 mons for a 4-node Ceph cluster. (With 4 mons, if any two nodes go down the cluster can't quorate. With 3, if two nodes go down but one of them is the node without a mon, which it will be 50% of the time, you've still got quorum.) 

Am I missing something that prevents this case? If not, can we add some logic with a set of workable mon counts (in our case, 3 and 5) and pick the largest one of those for which we have room? If for some reason the user really *wants* 4 mons, we should let the monCountOverrideEnvVar be authoritative, but most users will want an odd mon count.

Reply:
Yeah, the vars are bit of a misnomer actually. We only set odd counts, so currently only 3 and 5 are used.

Reply:
Ah, I see. The logic is a bit odd, but it works so long as we're only allowing min and max. Until that changes, no need to complicate things. Thanks Rohan.